### ISSUE TITLE
Testing brainstorming

### ISSUE THREAD

**Issue by nikipapadatou** [ethersphere/bee-backlog#73]
The following document focuses on a draft approach regarding the testing strategy. Each team of the organisation (where relevant) is called to brainstorm, discuss and get the appropriate tasks. Also, they are called to suggest improvements for the other teams whenever there is a dependency. 

https://hackmd.io/mUhde0eDRryKOjjUmN1huA?view

This is a placeholder task for the bee team, where all the members are called to express their opinions on what we should focus on the tests. Feel free to create a replicate document and write there your comments, improvements, ideas and the relevant tasks.

Please, attach your documents in the comments.

**Comment by mrekucci** [comment]
https://hackmd.io/@ethswarm/rkiiXGEbT

**Comment by ldeffenb** [comment]
> https://hackmd.io/@ethswarm/rkiiXGEbT

This might be ok in bee-backlog, but the main bee repository issues?

![image](https://github.com/ethersphere/bee/assets/5175026/00a3bebc-dacd-42db-a9ad-3a2c2086a7fe)

**Comment by tmm360** [comment]
I agree, please don't use references to private contents on public repos.
Actually, I would like to access documents also in the backlog area, anyway...

**Comment by janos** [comment]
My contribution to the testing brainstorming https://hackmd.io/SfY2ukFlSrqW6cqEv0w05Q?view.

**Comment by notanatol** [comment]
(updated) https://hackmd.io/7FDkHvB7QYi4MmgqNAgmCg

**Comment by ldeffenb** [comment]
> https://hackmd.io/@F-mGf2lBSoGylLGQE6rRkA/B1MSazfST

![image](https://github.com/ethersphere/bee-backlog/assets/5175026/1fabfe7f-8345-4627-9a9b-e58885697753)

**Comment by 0xCardiE** [comment]
As we don't have a clear path on how to move/arrive to swarm Testnet I will try to establish it now.
I would agree that testnet should be treated as a place for deploying code that is expected to fully work and ideally isn't having any problems. Hence, as the last line of "defense" before mainet and release, not something where we iterate and try things out.

Having said that we do need to try out smart contracts integrated with bee nodes somewhere.
So as we built this new hardhat replacement for docker, what  Vandot and I talked about is that we then make a cluster using that with geth node where contracts are deployed and some rounds are played(few days?).
Calling this "Pretesnet" for lack of better words, feel free to suggest better.
With this we would try out contracts with bee nodes, and see if deployment and game are running smoothly, after checking this and signing off by both teams we can go to the next step and add this to Testnet.

**Comment by ldeffenb** [comment]
IIRC, there are config parameters to override the hard-coded contract addresses for a bee node at runtime.   Could those be used, along with an appropriate bootnode, to allow folks outside the cluster to join the contract testing before formal testnet deployment?   Or is the cluster 100% self-contained with its own private blockchain and not actually deploying the test contracts on sepolia?
